jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/PythonGeneratorTranslator.java

```diff
@@ -531,204 +548,197 @@ private static void generateAdvanceGeneratorMethodForYieldFrom(ClassWriter class
-
-                methodVisitor.visitLdcInsn(1);
-                methodVisitor.visitJumpInsn(Opcodes.GOTO, iterateSubiterator);
-
-                methodVisitor.visitLabel(wasNotSentValue);
-
-                // Check if thrown a value
-                methodVisitor.visitVarInsn(Opcodes.ALOAD, 0);
-                methodVisitor.visitFieldInsn(Opcodes.GETFIELD, Type.getInternalName(PythonGenerator.class), "thrownValue",
-                        Type.getDescriptor(Throwable.class));
-                methodVisitor.visitInsn(Opcodes.ACONST_NULL);
-
-                methodVisitor.visitJumpInsn(Opcodes.IF_ACMPEQ, wasNotThrownValue);
-
-                methodVisitor.visitLdcInsn(2);
-                methodVisitor.visitJumpInsn(Opcodes.GOTO, iterateSubiterator);
-
-                methodVisitor.visitLabel(wasNotThrownValue);
-
-                methodVisitor.visitLdcInsn(0);
-
-                methodVisitor.visitLabel(iterateSubiterator);
-
-                Label tryStartLabel = new Label();
-                Label tryEndLabel = new Label();
-                Label catchStartLabel = new Label();
-                Label catchEndLabel = new Label();
-
-                methodVisitor.visitTryCatchBlock(tryStartLabel, tryEndLabel, catchStartLabel,
-                        Type.getInternalName(StopIteration.class));
-
-                methodVisitor.visitLabel(tryStartLabel);
-                BytecodeSwitchImplementor.createIntSwitch(methodVisitor, List.of(0, 1, 2),
-                        key -> {
-                            Label generatorOperationDone = new Label();
-                            switch (key) {
-                                case 0: { // next
-                                    DunderOperatorImplementor.unaryOperator(methodVisitor, PythonUnaryOperator.NEXT);
-                                    break;
-                                }
-                                case 1: { // send
-                                    methodVisitor.visitVarInsn(Opcodes.ALOAD, 0);
-                                    methodVisitor.visitFieldInsn(Opcodes.GETFIELD, Type.getInternalName(PythonGenerator.class),
-                                            "sentValue",
-                                            Type.getDescriptor(PythonLikeObject.class));
+        PythonBytecodeToJavaBytecodeTranslator.writeInstructionsForOpcodes(generatorMethodPart.functionMetadata,
+                stackMetadataForOpcodeIndex, opcodeList,
+                instruction -> {
+                    if (instruction.offset == generatorMethodPart.afterYield) {
+                        // 0 = next, 1 = send, 2 = throw
+
+                        Label wasNotSentValue = new Label();
+                        Label wasNotThrownValue = new Label();
+                        Label iterateSubiterator = new Label();
+
+                        // Push subiterator
+                        methodVisitor.visitVarInsn(Opcodes.ALOAD, 0);
```

Comment:
As much as I appreciate that `instruction` is a long word and there will be a lot of use, I don't think that the "insn" abbreviation is particularly easy to decipher.
Also, as other parts of the code clearly use the word "instruction" (see line 561), I'd vote to just use "instruction" across the board for consistency.

Reply:
This is from ASM, the Java bytecode code generation we (and Gizmo) use. See section 3.1.2 of the ASM user guide (https://asm.ow2.io/asm4-guide.pdf) for an in depth explanation of how this works. For this particular instruction, this loads the object reference stored at slot 0, which for virtual methods (which this IS a virtual method), is always `this`.